staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2233,6 +2243,11 @@ type Container struct {
 	// Default is false.
 	// +optional
 	TTY bool `json:"tty,omitempty" protobuf:"varint,18,opt,name=tty"`
+	//Container Lifecycle.
+	//One of Standard, Sidecar.
+	//Defaults to Standard
+	//+optional
+	ContainerLifecycle ContainerLifecycle `json:"containerLifecycle,omitempty" protobuf:"bytes,22,opt,name=containerLifecycle,casttype=ContainerLifecycle"`
```

Comment:
Having `pod.spec.containers.lifecycle` and `pod.spec.containers.containerLifecycle` might be confusing to users. Have you thought about introducing `pod.spec.containers.lifecycle.type` instead?

Reply:
yeah I did consider that, but I thought it might be confusing as the existing things in `lifecycle` are actions to perform whereas this isn't an action, it's a way of describing when your container should start up/shut down. There's a PR here https://github.com/kubernetes/enhancements/pull/919 where the API implementation is being discussed